include/hal/library/memlib.h

```diff
@@ -87,27 +87,19 @@ void *libspdm_set_mem(void *buffer, size_t length, uint8_t value);
 void *libspdm_zero_mem(void *buffer, size_t length);
 
 /**
- * Compares the contents of two buffers in const time.
+ * Compares the contents of two buffers in constant time.
  *
- * This function compares length bytes of source_buffer to length bytes of destination_buffer.
- * If all length bytes of the two buffers are identical, then 0 is returned.  Otherwise, the
- * value returned is the first mismatched byte in source_buffer subtracted from the first
- * mismatched byte in destination_buffer.
+ * For a given length, the time to complete the comparison is always the same regardless of the
+ * contents of the two buffers.
  *
- * If length > 0 and destination_buffer is NULL, then ASSERT().
- * If length > 0 and source_buffer is NULL, then ASSERT().
- * If length is greater than (MAX_ADDRESS - destination_buffer + 1), then ASSERT().
- * If length is greater than (MAX_ADDRESS - source_buffer + 1), then ASSERT().
- *
- * @param  destination_buffer A pointer to the destination buffer to compare.
- * @param  source_buffer      A pointer to the source buffer to compare.
- * @param  length            The number of bytes to compare.
- *
- * @return 0                 All length bytes of the two buffers are identical.
- * @retval Non-zero          There is mismatched between source_buffer and destination_buffer.
+ * @param  destination_buffer  A pointer to the destination buffer to compare.
+ * @param  source_buffer       A pointer to the source buffer to compare.
+ * @param  length              The number of bytes to compare.
  *
+ * @return true   The contents of the two buffers are the same.
+ * @retval false  The contents of the two buffers are not the same.
  **/
-int32_t libspdm_const_compare_mem(const void *destination_buffer,
-                                  const void *source_buffer, size_t length);
+bool libspdm_const_compare_mem(const void *destination_buffer,
```

Comment:
Why we make it different from std `memcmp` ?

I don't understand why we have to make this change.

Reply:
Resolved in original issue.